### ISSUE TITLE
for_window [floating] does not update except on exit to a program and may result in incorrect title

### ISSUE THREAD

**Issue by rieje** [swaywm/sway#6606]
- **Sway Version:**
  - 1.6.1

- **Configuration File:**
  - [minimal config https://pastebin.com/raw/LAz7MAcB]

- **Description:**

Window title is never updated correctly and the behaviors are different depending on apps that change the window title(?). With the following rules:

    for_window [floating] title_format "%title [floating]"
    for_window [tiling] title_format "%title"

1) Execute keybinding to start terminal in tiling, window title is "`Alacritty`". 
2) Make it floating, window title is still "`Alacritty`" (expect "`Alacritty [floating]`").
3) Launch `nvim` in terminal, window title is "`Alacritty`".
4) Exit nvim and window title becomes "`Alacritty [floating]`".
5) Make floating window back to tiling, window title is still "`Alacritty [floating]`" (expect "`Alacritty`").

Different behavior with another app:

1) Execute keybinding to start terminal in tiling, window title is "`Alacritty`". 
2) Make it floating, window title is still "`Alacritty`" (expect "`Alacritty [floating]`").
3) Launch `ncmpcpp` in terminal, window title is "`my music [floating]`".
4) Exit nvim and window title becomes "`[floating]`". <--- Should be "`Alacritty [floating]`" like it is in step 4 with nvim above?
5) Make floating window back to tiling, window title is still "`[floating]`"  (expect "`Alacritty`").

Simply toggling window between floating and tiling never changes the title.

- **Goal:**
Modify `title_format` of window titles of floating windows to better visually indicate whether the focused window is tiling or floating. Also, adding a suffix like `[floating]` to the window name allows Waybar to match to window names of floating windows and stylize them on the status bar (e.g. color the text of the window name green) make it more obvious current focus is on floating window.

**Comment by asiantuntija** [comment]
Yeah, I can confirm this, just spent half on hour wondering why nothing was happening when making windows floating with `for_window [floating]`.

Executing `swaymsg -t get_tree | grep floating` shows that when making a window float it changes type immediately to `floating_con`, but as @rieje said, matching to floating rules happens later, for example Alacritty gets matched after running a command. I couldn't get matching to work at all on Gnome Calculator main window, only windows spawned by it get matched.

Toggling the window back to tiled and `for_window [tiling]` doesn't get matched.
